### PR TITLE
temporarily restrict /gene_selection/ pool to CGC genes

### LINKED ISSUE #884: gdc /gene_selection/ restrict to CGC set
temporarily

### PR DESCRIPTION
**Pull request by xzhou82**
## Description

fixes #884 
test at http://localhost:3000/example.gdc.exp.html

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: server/routes/gdc.topVariablyExpressedGenes.ts [hunk 1]
```diff
@@ -83,9 +79,14 @@ async function getGenes(q: any, ds: any, genome: any) {
...
-				gene_type: 'protein_coding',
-				selection_size: Number(q.maxGenes || 100)
+
+				// temporary!! restrict pool to cgc due to slow api. delete when new api is online
+				gene_ids: tempGetCGCgenes(genome),
+
+				// when gene_ids is deleted, enable this
+				//gene_type: 'protein_coding',
+
+				selection_size: Number(q.maxGenes) // FIXME it's defined as number but why it's string??
```
Comment (xzhou82) [inline server/routes/gdc.topVariablyExpressedGenes.ts:89]: q.maxGenes value is still string but it's declared as number

File: server/routes/gdc.topVariablyExpressedGenes.ts [hunk 2]
```diff
@@ -111,3 +112,17 @@ async function getGenes(q: any, ds: any, genome: any) {
...
 }
+
+function tempGetCGCgenes(genome: any) {
+	const lst = [] as string[] // list of ENSG ids from cgc genes
+	// don't think there's need to preparse genome.geneset, as this function is only temporary
+	for (const s of genome.geneset[0].lst) {
+		const a = genome.genedb.getAliasByName.all(s)
+		if (a) {
+			for (const b of a) {
+				if (b.alias.startsWith('ENSG')) lst.push(b.alias)
```
Comment (siosonel) [inline server/routes/gdc.topVariablyExpressedGenes.ts:123]: @xzhou82 Does this mean that a non-CGC genes would not have a `ENSG` alias? In other words, only CGC genes are expected to have `ENSG` aliases? Just asking for my own information.

**Comment by xzhou82** [comment]
all genes have gencode accession (ENSG)

**Review by siosonel — APPROVED** [review]

**Comment by siosonel** [comment]
@xzhou82 I updated the default term group name in `sjpp/gdc/active/dataset/gdc.hg38.ts` master branch. That update, plus my recent commit in this branch, is required in order to see the `Gene Expression (CGC genes only)` label.

**Pull request merged**